src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -122,7 +122,8 @@ protected virtual DefaultMetadataDetails[] CreatePropertyDetails([NotNull] Model
                 var propertyEntry = new DefaultMetadataDetails(propertyKey, attributes);
                 if (propertyHelper.Property.CanRead && propertyHelper.Property.GetMethod?.IsPublic == true)
                 {
-                    propertyEntry.PropertyGetter = propertyHelper.ValueGetter;
+                    var getter = PropertyHelper.MakeNullSafeFastPropertyGetter(propertyHelper.Property);
+                    propertyEntry.PropertyGetter = getter;
```

Comment:
Will add a test that actually uses the accessors.

As for propertyhelper, I don't see a reason to change it. In the validation/binding code we have a good reason to care about null-safety. In general it's something I'd avoid.